components/markets/SellFullSetModal.tsx

```diff
@@ -1,139 +1,179 @@
-            balance.lessThan(lowestTokenAmount)
-          ) {
-            lowestTokenAmount = balance;
-          }
-        }
-        setMaxTokenSet(lowestTokenAmount ?? new Decimal(0));
-      });
-    }, [marketStore]);
+import { useEffect, useState } from "react";
+import Loader from "react-spinners/PulseLoader";
+
+const SellFullSetModal = observer(({ marketId }: { marketId: number }) => {
+  const store = useStore();
+  const { wallets } = store;
+  const notificationStore = useNotificationStore();
+  const modalStore = useModalStore();
+
+  const { data: market } = useMarket(marketId);
+  const { data: saturatedMarket } = useSaturatedMarket(market);
+  const { data: pool } = usePool({ marketId: marketId });
+
+  const { data: balances } = useAccountPoolAssetBalances(
+    wallets.getActiveSigner(),
+    pool,
+  );
+
+  const [transacting, setTransacting] = useState(false);
+  const [amount, setAmount] = useState<string>("0");
+  const [maxTokenSet, setMaxTokenSet] = useState<Decimal>(new Decimal(0));
 
-    const handleAmountChange = (amount: string) => {
-      setAmount(amount);
-    };
+  useEffect(() => {
+    let lowestTokenAmount: Decimal = null;
+    balances?.forEach((balance) => {
+      const free = new Decimal(balance.free.toNumber());
+      if (!lowestTokenAmount || free.lessThan(lowestTokenAmount)) {
+        lowestTokenAmount = free;
+      }
+    });
+    setMaxTokenSet(lowestTokenAmount ?? new Decimal(0));
+  }, [balances]);
 
-    const handleSignTransaction = async () => {
-      if (Number(amount) > maxTokenSet.toNumber() || Number(amount) === 0)
-        return;
+  const handleAmountChange = (amount: string) => {
+    setAmount(amount);
+  };
+
+  const handleSignTransaction = async () => {
+    if (
+      Number(amount) > wallets.activeBalance.toNumber() ||
+      Number(amount) === 0
+    ) {
+      return;
+    }
+
+    setTransacting(true);
+
+    if (hasMarketMethods<Context>(market)) {
       const signer = wallets.getActiveSigner();
 
-      await marketStore.market.sellCompleteSet(
-        signer,
-        new Decimal(amount).mul(ZTG).toNumber(),
-        extrinsicCallback({
-          notificationStore,
-          successCallback: () => {
-            notificationStore.pushNotification(
-              `Sold ${new Decimal(amount).toFixed(1)} full sets`,
-              { type: "Success" },
-            );
-            modalStore.closeModal();
-            marketsStore.getMarket(marketStore.id);
-          },
-          failCallback: ({ index, error }) => {
-            notificationStore.pushNotification(
-              store.getTransactionError(index, error),
-              {
-                type: "Error",
-              },
-            );
+      notificationStore?.pushNotification("Transacting...", {
+        autoRemove: true,
+      });
+
+      const result = await market
+        .sellCompleteSet({
+          amount: new Decimal(amount).mul(ZTG).toNumber(),
+          signer,
+          hooks: {
```

Comment:
We have most of this logic held in `extrinsicCallback` is it possible to reuse that? I think this sort of behaviour should be consistent throughout the app. I also prefer the behaviours provided by `extrinsicCallback` such as a notification displayed after signing and a success message once the transaction is in block

Reply:
Made an issue for a refactoring for this. Have to think about the best approach for it